File: Stage31/C/Problem2293.c
#include <stdio.h>

int main(void) {
    int n, k;
    scanf("%d %d", &n, &k);

    int coins[100];
    for (int i = 0; i < n; i++) {
        scanf("%d", &coins[i]);
    }

    int dp[10001] = {0};
    dp[0] = 1;

    for (int i = 0; i < n; i++) {
        int coin = coins[i];
        for (int j = coin; j <= k; j++) {
            dp[j] += dp[j - coin];
        }
    }

    printf("%d\n", dp[k]);
    return 0;
}

File: Stage31/Python/Problem2293.py
import sys

input = sys.stdin.readline

def main():
    n, k = map(int, input().split())
    coins = [int(input()) for _ in range(n)]

    dp = [0] * (k + 1)
    dp[0] = 1

    for coin in coins:
        for i in range(coin, k + 1):
            dp[i] += dp[i - coin]

    print(dp[k])

if __name__ == "__main__":
    main()